### PR TITLE
Alternator measuring RCU and WCU

### PR DESCRIPTION
**Pull request by amnonh**
Read and Write Consumed Capacity units are an abstract way of measuring Alternator actions. In general, they correspond to the read or write data.

In the long run, the RCU/WCU adds a way of charging an operation and limiting usage.

This series addresses two issues: consume capacity request API and metering.

The Alternator (and DynmoDB) API has an optional parameter allowing users to check the number of units an operation consumes. When a user adds that parameter, the response will contain the number of units used for the operation.

This series adds the consume capacity support to the get_item and put_item, adds a metric to collect the overall RCU and WCU used, and adds a test for the new functionality.

Follow-up PRs will add support for more operations and GSI.

Replaces #19811
Partially implement: #5027

### CODE REVIEW

**Comment by nyh** [comment]
Please mention in in the cover letter that this patch partially implemented https://github.com/scylladb/scylladb/issues/5027 - and when this PR is merged, you should explain in 5027 what has been done and what is left (specifically, list the operations that were node done, and say that only TOTAL mode was implemented and INDEXES mode was not).

Moreover, please mention in the cover letter and 5027 (and if possible, write a test) whether or not the support in this code is correct for tables for GSI. I.e., a is write to a table with GSI counted the same way it is (with extra counting of the index table) as in DynamoDB?

File: test/alternator/test_returnconsumedcapacity.py [hunk 1]
```diff
@@ -6,10 +6,13 @@
...
 import pytest
 from botocore.exceptions import ClientError
+import decimal
 from decimal import Decimal
-from test.alternator.util import random_string, random_bytes
+import boto3.dynamodb.types
+from test.alternator.util import random_string, random_bytes, new_test_table
 
 KB = 1024
+boto3.dynamodb.types.DYNAMODB_CONTEXT = decimal.Context(prec=100)
```
[the remark this answers is not in the document]
Comment (nyh) [reply test/alternator/test_returnconsumedcapacity.py:15]: Created https://github.com/scylladb/scylladb/pull/21635 to fix the source of your copy :-)

[the remark this answers is not in the document]
Comment (nyh) [reply test/alternator/test_returnconsumedcapacity.py:15]: By the way, is this monkey-patching really necessary in this test? I thought I remembered it is needed for reading back a long number - but not for writing it initially. But maybe I'm misremembering.

**Review by nyh — APPROVED** [review]
Looks mostly good to me so I'm tentatively "approving". But I left a few comments/requests behind, please take a look.

File: alternator/executor.cc [hunk 2]
```diff
@@ -1535,6 +1535,7 @@ class put_or_delete_item {
     // PutItem: engaged _cells, write these cells to item (_pk, _ck).
     // DeleteItem: disengaged _cells, delete the entire item (_pk, _ck).
     std::optional<std::vector<cell>> _cells;
+    uint64_t _message_length_in_bytes = 0;
```
[the remark this answers is not in the document]
Comment (amnonh) [reply alternator/executor.cc:1538]: the memeber used for all implementation, it's an open issue what we'll do about delete

File: alternator/executor.cc [hunk 3]
```diff
@@ -1545,6 +1546,9 @@ class put_or_delete_item {
     mutation build(schema_ptr schema, api::timestamp_type ts) const;
     const partition_key& pk() const { return _pk; }
     const clustering_key& ck() const { return _ck; }
+    uint64_t length_in_bytes() const noexcept {
+        return _message_length_in_bytes;
+    }
```
[the remark this answers is not in the document]
Comment (amnonh) [reply alternator/executor.cc:1551]: I'll make it consistent and add a comment

File: test/alternator/test_returnconsumedcapacity.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (amnonh) [reply test/alternator/test_returnconsumedcapacity.py]: I've minimize my import and was able to drop that patching

**Comment by nyh** [comment]
Thanks! Finally merged.

**Pull request merged**